### ISSUE TITLE
Installer doesn't update language definitions

### ISSUE THREAD

**Issue by afalkenhahn** [notepad-plus-plus/notepad-plus-plus#14027]
### Description of the Issue
The installer doesn't update the files `langs.xml` and `stylers.xml` in `%APPDATA%\Roaming\Notepad++` which means that newly added language definitions won't be recognized at all by Notepad++. Furthermore, the file `%PROGRAMFILES%\Notepad++\functionList\overrideMap.xml` isn't updated by the installer either which means that the function lister won't work for newly added language definitions.
 
### Steps to Reproduce the Issue
1. Install Notepad++ 8.5
2. Now install the latest Notepad++ version (currently 8.5.6) on top of it

### Expected Behavior
I'd expect all languages added after 8.5 (e.g. MSSQL, GDScript, Hollywood) to be working.

### Actual Behavior
Notepad++ 8.5.6 doesn't recognize the newly added languages because it's still using the definition files from Notepad++ 8.5 because the installer doesn't update the files.

### Workaround
The issue can be worked around by completely uninstalling the old Notepad++ version before installing the new one but that is rather inconvenient of course. The installer should automatically update the language definition files when installing a new version.

**Comment by victorel-petrovich** [comment]
Related: https://github.com/notepad-plus-plus/notepad-plus-plus/issues/13932

**Comment by afalkenhahn** [comment]
Note that when I say "language" in my report, I'm talking about programming languages, not about human languages. So in contrast to #13932 my issue is not in any way related to localization, just to make this clear.

**Comment by victorel-petrovich** [comment]
@afalkenhahn , you're right, I'll delete my comment , you can delete yours.

**Comment by schnaggi** [comment]
No, the installer only should update the locatizations in application folder. The app itself should update the files in userprofile if required.
Reason: It is a machine installer which may not update files in user profiles.

**Comment by afalkenhahn** [comment]
> No, the installer only should update the locatizations in application folder. The app itself should update the files in userprofile if required. Reason: It is a machine installer which may not update files in user profiles.

@schnaggi I think there is a misunderstanding here. I'm not talking about languages as in English, German and French but about programming languages like C, Java, and Python.

Think about the following case: The user installs Notepad++ 8.5. Later he installs 8.5.6 which adds support for programming languages like MSSQL, GDScript, Hollywood. Although the user has installed 8.5.6 now, Notepad++ still won't handle those newly added programming languages correctly because the installer doesn't update `langs.xml` and `stylers.xml` in `%APPDATA%\Roaming\Notepad++`. The only way to make Notepad++ handle the new programming languages correctly is to uninstall the old Notepad++ version first. I think this is a major issue and should definitely be fixed!

**Comment by ScrubN** [comment]
I use notepad++ weekly with several different languages. Recently I got fed up with it not automatically recognizing recently added formats like gdscript, go, etc. and found this issue that seems to describe exactly what is wrong.

> Edit: and the files are updated if you open language options and reselect the used language, IIRC

My `langs.xml` file has a last modified date of 2020 (when I install np++ on this machine) and lacks entries for the previously mentioned languages that are not automatically recognized, despite manually changing the language setting as you say.


If there is any other way to manually update `langs.xml` and `stylers.xml`, please let me know. It's really annoying to manually select the file language every time I open a file.

**Comment by alankilborn** [comment]
> any other way to manually update `langs.xml` and `stylers.xml`, please let me know

See https://community.notepad-plus-plus.org/topic/26049

**Comment by ScrubN** [comment]
> See https://community.notepad-plus-plus.org/topic/26049

Thank you, this helped a lot. Newly added file types are supported now and more syntax tokens are highlighted now.

I feel like this functionality should be built into np++ though, since many users may not realize that this is an issue and would instead assume np++ is just broken.

**Comment by alankilborn** [comment]
> I feel like this functionality should be built into np++ though, since many users may not realize that this is an issue and would instead assume np++ is just broken.

Well, this issue is currently "Open", so the Notepad++ developer (or an interested developer) could implement...
